service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -1476,6 +1493,17 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
                 + "because it has been concurrently modified to %s",
             tableIdentifier, oldLocation, newLocation, existingLocation);
       }
+
+      // We diverge from `BaseMetastoreTableOperations` in the below code block
+      if (updateMetadataOnCommit) {
```

Comment:
"update" sounds very intrusive to me, and it feel related to the metadata in storage... How about `makeMetadataCurrentOnCommit`?